core/src/test/scala/org/apache/spark/shuffle/unsafe/UnsafeShuffleSuite.scala

```diff
@@ -41,6 +42,25 @@ class UnsafeShuffleSuite extends ShuffleSuite with BeforeAndAfterAll {
     conf.set("spark.shuffle.memoryFraction", "0.5")
   }
 
+  override def multipleAttemptConfs: Seq[(String, SparkConf)] = {
+    // unsafe shuffle has a few different code paths, based on various configs.  We want to
+    // make sure we stress multiple attempts under all variants
+    val kryoAndNoSpillMove = conf.clone()
+      .set("spark.shuffle.manager", "tungsten-sort")
+      .set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .set("spark.shuffle.unsafe.testing.allowSpillMove", "false")
+    val noCompressionFileMerge = kryoAndNoSpillMove.clone()
+      .set("spark.shuffle.compress", "false")
+      .set("spark.file.transferTo", "false")
+    val noCompressionTransferTo = noCompressionFileMerge.clone()
+      .set("spark.file.transferTo", "true")
+    Seq(
+      "slow merge path" -> kryoAndNoSpillMove,
+      "filestream based fast merge" -> noCompressionFileMerge,
+      "transferTo based fast merge" -> noCompressionTransferTo
```

Comment:
suggestions on other confs I should include here?  this list is just from a scan of the unsafe shuffle code ...